app/src/main/java/com/humberto/tasky/core/data/AndroidConnectivityObserver.kt

```diff
@@ -0,0 +1,53 @@
+package com.humberto.tasky.core.data
+
+import android.net.ConnectivityManager
+import android.net.ConnectivityManager.NetworkCallback
+import android.net.Network
+import android.net.NetworkCapabilities
+import com.humberto.tasky.core.domain.ConnectivityObserver
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import javax.inject.Inject
+
+class AndroidConnectivityObserver @Inject constructor(
+    private val connectivityManager: ConnectivityManager
+): ConnectivityObserver {
+
+    override val isConnected: Flow<Boolean>
+        get() = callbackFlow {
+            val callback = object : NetworkCallback() {
+                override fun onAvailable(network: Network) {
+                    super.onAvailable(network)
+                    trySend(true)
```

Comment:
Philipp released a new video on monitoring network connection status, which I think you’ll find helpful: https://www.youtube.com/watch?v=wvDPG2iQ-OE

The problem with the older approach is that it shows you only whether or not the device should be able to have internet connection (based on if wifi is enabled, etc.)

Reply:
That's the video I followed 😄 I'm overriding onCapabilitiesChanged just below